lisa/sut_orchestrator/hyperv/platform_.py

```diff
@@ -0,0 +1,323 @@
+            parent_logger=get_logger("hyperv"),
+        )
+
+        server_node.set_connection_info(
+            address=server.address, username=server.username, password=server.password
+        )
+
+        return server_node
+
+    def _prepare_environment(self, environment: Environment, log: Logger) -> bool:
+        return self._configure_node_capabilities(environment, log)
+
+    def _configure_node_capabilities(
+        self, environment: Environment, log: Logger
+    ) -> bool:
+        if not environment.runbook.nodes_requirement:
+            return True
+
+        nodes_capabilities = self._create_node_capabilities()
+
+        nodes_requirement = []
+        for node_space in environment.runbook.nodes_requirement:
+            if not node_space.check(nodes_capabilities):
+                return False
+
+            requirement = node_space.generate_min_capability(nodes_capabilities)
+            nodes_requirement.append(requirement)
+
+        environment.runbook.nodes_requirement = nodes_requirement
+        return True
+
+    def _init_host_capabilities(self, log: Logger) -> _HostCapabilities:
+        host_cap = _HostCapabilities()
+
+        free_mem_bytes = self.server_node.tools[PowerShell].run_cmdlet(
+            "(Get-CimInstance -ClassName Win32_OperatingSystem).FreePhysicalMemory"
+        )
+        host_cap.free_memory_mib = int(free_mem_bytes) // (1024 * 1024)
+
+        lp_count = self.server_node.tools[PowerShell].run_cmdlet(
+            "(Get-CimInstance Win32_ComputerSystem).NumberOfLogicalProcessors"
+        )
+        host_cap.core_count = int(lp_count)
+
+        log.debug(
+            f"Host capabilities: {host_cap.core_count} cores, "
+            f"{host_cap.free_memory_mib} MiB free memory"
+        )
+
+        return host_cap
+
+    # Check that the VM requirements can be fulfilled by the host.
+    def _check_host_capabilities(
+        self,
+        nodes_requirements: List[schema.NodeSpace],
+        host_capabilities: _HostCapabilities,
+        log: Logger,
+    ) -> bool:
+        total_required_memory_mib = 0
+
+        for node_requirements in nodes_requirements:
+            # Calculate the total amount of memory required for all the VMs.
+            assert isinstance(node_requirements.memory_mb, int)
+            total_required_memory_mib += node_requirements.memory_mb
+
+        # Ensure host has enough memory for all the VMs.
+        if total_required_memory_mib > host_capabilities.free_memory_mib:
+            log.error(
+                f"Nodes require a total of {total_required_memory_mib} MiB memory. "
+                f"Host only has {host_capabilities.free_memory_mib} MiB free."
+            )
+            return False
+
+        return True
+
+    def _create_node_capabilities(self) -> schema.NodeSpace:
+        host_capabilities = self._host_capabilities
+        node_capabilities = schema.NodeSpace()
+        node_capabilities.name = "Hyper-V"
```

Comment:
Should it be used as VM name? It should be set to empty, or use an unique value, when vm is created.

Reply:
This is not per VM. Baremetal platform is also using a fixed name "baremetal".

Reply:
use hyperv for all lower case. I will look at it later.

Reply:
changed.